src/nvim/eval.c

```diff
@@ -14158,32 +14189,28 @@ static void f_rpcstart(typval_T *argvars, typval_T *rettv)
 
   // The last item of argv must be NULL
   argv[i] = NULL;
-  uint64_t channel_id = channel_from_process(argv);
 
-  if (!channel_id) {
-    EMSG(_(e_api_spawn_failed));
-  }
-
-  rettv->vval.v_number = (varnumber_T)channel_id;
+  TerminalJobData *data = common_job_init(argv, NULL, NULL, NULL,
+                                          NULL, false, true, false, NULL);
+  common_job_start(data, rettv);
 }
 
-// "rpcstop()" function
+// "rpcstop()" function (DEPRECATED)
```

Comment:
I'll remove the deprecation. We cannot fully deprecate it yet as it is the only way to close a socket (which currently can only be opened from outside nvim, so I don't expect it to be used much yet ). We should fix it in a further PR allowing full socket IO.